File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Model/Pokemon/Coin.java
package com.pokeapi.lpiem.pokeapiandroid.Model.Pokemon;

public class Coin{
    private Element coinElementType;
    private String coinFileUrl;

    /**
     * Constructeur par défaut
     */
    public Coin(){
        coinElementType = new Element();
        coinFileUrl = "";
    }

    public Coin(String coinFileUrl, Element coinElementType){
        this.coinElementType = coinElementType;
        this.coinFileUrl = coinFileUrl;
    }

    public Element getCoinElementType(){
        return coinElementType;
    }

    public String getCoinFileUrl(){
        return this.coinFileUrl;
    }
}

File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Model/Pokemon/Retrofit/Coin.kt
package com.pokeapi.lpiem.pokeapiandroid.Model.Pokemon.Retrofit

class Coin {
    var coinElementType: Element? = null
        private set
    var coinFileUrl: String? = null
        private set

    /**
     * Constructeur par défaut
     */
    constructor() {
        coinElementType = Element()
        coinFileUrl = ""
    }

    constructor(coinFileUrl: String, coinElementType: Element) {
        this.coinElementType = coinElementType
        this.coinFileUrl = coinFileUrl
    }
}